ansible/roles/test/files/ptftests/inner_hash_test.py

```diff
@@ -0,0 +1,505 @@
+        @param src_port: index of port to use for sending packet to switch
+        @param ip_dst: Inner packet Destination IP address
+        @param ip_src: Inner packet Source IP address
+        @param dport: Inner packet dst port
+        @param sport: Inner packet src port
+        @param ip_proto: Inner packet ip protocol
+        '''
+        src_mac = self.dataplane.get_mac(0, src_port)
+
+        outer_ip_src = self.outer_src_ip_interval.get_random_ip() if hash_key == 'outer-tuples' else self.outer_src_ip_interval.get_first_ip()
+        outer_ip_dst = self.outer_dst_ip_interval.get_random_ip() if hash_key == 'outer-tuples' else self.outer_dst_ip_interval.get_first_ip()
+        outer_sport = random.randint(0, 65535) if hash_key == 'outer-tuples' else 1234
+
+        pkt = self.generate_inner_pkt(sport, dport, ip_src, ip_dst, ip_proto)
+        vxlan_pkt = simple_vxlanv6_packet(
+                    eth_dst=self.router_mac,
+                    eth_src=src_mac,
+                    ipv6_src=outer_ip_src,
+                    ipv6_dst=outer_ip_dst,
+                    udp_sport=outer_sport,
+                    udp_dport=4789,
+                    vxlan_vni=random.randint(1, 254)+20000,
+                    with_udp_chksum=False,
+                    inner_frame=pkt)
+
+        logging.info("Sending packet from port {} outer_ip_src {} outer_ip_dst {} inner_src_ip {} inner_dst_ip {} inner_sport {} inner_dport {} inner_ipproto {}"\
+                .format(src_port, outer_ip_src, outer_ip_dst, ip_src, ip_dst, sport, dport, ip_proto))
+        send_packet(self, src_port, vxlan_pkt)
+
+        masked_exp_pkt = Mask(vxlan_pkt)
+        masked_exp_pkt.set_do_not_care_scapy(scapy.Ether, "dst")
+        masked_exp_pkt.set_do_not_care_scapy(scapy.Ether, "src")
+        masked_exp_pkt.set_do_not_care_scapy(scapy.IPv6, "hlim")
+
+        return verify_packet_any_port(self, masked_exp_pkt, self.exp_port_list)
+
+
+    def check_within_expected_range(self, actual, expected):
+        '''
+        @summary: Check if the actual number is within the accepted range of the expected number
+        @param actual : acutal number of recieved packets
+        @param expected : expected number of recieved packets
+        @return (percentage, bool)
+        '''
+        percentage = (actual - expected) / float(expected)
+        return (percentage, abs(percentage) <= self.balancing_range)
+
+
+    def check_balancing(self, dest_port_list, port_hit_cnt):
+        '''
+        @summary: Check if the traffic is balanced across the ECMP groups and the LAG members
+        @param dest_port_list : a list of ECMP entries and in each ECMP entry a list of ports
+        @param port_hit_cnt : a dict that records the number of packets each port received
+        @return bool
+        '''
+
+        logging.info("%-10s \t %-10s \t %10s \t %10s \t %10s" % ("type", "port(s)", "exp_cnt", "act_cnt", "diff(%)"))
```

Comment:
@anish-n please add some more logging here to explain what will be printed below

Reply:
Its a tabular format to show the exp_cnt, actual_cnt etc, basically what the logic verifies in terms of deviation between exp_cnt and act_cnt is listed via log statement. I think this part is explained via the log statement already?